irobotclient/main.py

```diff
@@ -0,0 +1,100 @@
+"""
+Copyright (c) 2017 Genome Research Ltd.
+
+This program is free software: you can redistribute it and/or modify it
+under the terms of the GNU General Public License as published by the
+Free Software Foundation, either version 3 of the License, or (at your
+option) any later version.
+
+This program is distributed in the hope that it will be useful, but
+WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+Public License for more details.
+
+You should have received a copy of the GNU General Public License along
+with this program. If not, see <http://www.gnu.org/licenses/>.
+"""
+import errno
+
+from os import path
+from requests import Response
+
+from irobotclient import configuration_handler
+from irobotclient import request_formatter
+from irobotclient.custom_exceptions import IrobotClientException
+from irobotclient.request_handler import Requester
+
+
+def _print_error_details(error: OSError):
+    """
+    Print the output of any errors raised.
+
+    Utilises the OSError exception class which the iRobot Client uses for custom exceptions too.
+
+    :param error:
+    :return:
+    """
+    print("Inside _output_error_details") # Beth - debug
+    print(error)
+    exit(error.errno)
+
+
+def _download_data(response: Response, output_dir:str):
+    """
+
+    :param response:
+    :param output_dir:
+    :return:
+    """
+    try:
+        file_name = (path.split(response.url))[1]
```

Comment:
As discussed some time ago, the idea was to have a command line option that would do something sensible in this event. For example, it could rename the output files such they don't collide, in some deterministic way:

iRODS Path | Output Filename
-- | --
/path/to/my/file | %path%to%my%file
/path/to/some/other/file | %path%to%some%other%file

Either that, or you could replicate the (minimal) directory structure in the output directory.

If that command line option isn't set and there is a collision, then it should raise some kind of error (which can be detected before you start trying to download).